content/blog/2017/2017-02-14-declarative-notifications.adoc

```diff
@@ -0,0 +1,436 @@
+
+  // Override default values based on build status
+  if (buildStatus == 'STARTED') {
+    color = 'YELLOW'
+    colorCode = '#FFFF00'
+  } else if (buildStatus == 'SUCCESSFUL') {
+    color = 'GREEN'
+    colorCode = '#00FF00'
+  } else {
+    color = 'RED'
+    colorCode = '#FF0000'
+  }
+
+  // Send notifications
+  slackSend (color: colorCode, message: summary)
+
+  hipchatSend (color: color, notify: true, message: summary)
+
+  emailext (
+      to: 'bitwiseman@bitwiseman.com',
+      subject: subject,
+      body: details,
+      recipientProviders: [[$class: 'DevelopersRecipientProvider']]
+    )
+}
+----
+
+.vars/sendNotifications.txt
+[source]
+----
+<strong>sendNotifications(buildStatus='STARTED')</strong>
+
+<p>
+    Sends notifications via HipChat, Slack, and email.
+    Defaults the sending "Started" notification.
+    Pass the build status string and it customize the notifications based on status.
+</p>
+----
+
+.Jenkinsfile
+[pipeline]
+----
+// Declarative //
+#!groovy
+@Library('bitwiseman-shared@blog/declarative/notifications') _
+
+pipeline {
+  agent {
+    // Use docker container
+    docker {
+      image 'ruby:2.3'
+    }
+  }
+  options {
+    // Only keep the 10 most recent builds
+    buildDiscarder(logRotator(numToKeepStr:'10'))
+  }
+  stages {
+    stage ('Start') {
+      steps {
+        // send build started notifications
+        sendNotifications 'STARTED'
+      }
+    }
+    stage ('Install') {
+      steps {
+        // install required bundles
+        sh 'bundle install'
+      }
+    }
+    stage ('Build') {
+      steps {
+        // build
+        sh 'bundle exec rake build'
+      }
+
+      post {
+        success {
+          // Archive the built artifacts
+          archive includes: 'pkg/*.gem'
+        }
+      }
+    }
+    stage ('Test') {
+      steps {
+        // run tests with coverage
+        sh 'bundle exec rake spec'
+      }
+
+      post {
+        success {
+          // publish html
+          publishHTML target: [
+              allowMissing: false,
+              alwaysLinkToLastBuild: false,
+              keepAll: true,
+              reportDir: 'coverage',
+              reportFiles: 'index.html',
+              reportName: 'RCov Report'
+            ]
+        }
+      }
+    }
+  }
+  post {
+    always {
+      sendNotifications currentBuild.result
+    }
+  }
+}
+// Scripted //
+----
+
+
+== Conclusion
+In this post we added notifications to our Declarative Pipeline.
+I was pleased to see how little we had to disrupt the existing
+flow of our pipeline to do so. The changes were the start and
+end of the file with no reformatting elsewhere.
+Then with the help of the Shared Library feature,
+we've moved the implementation of `sendNotifications` out of
+`Jenkinsfile`.  This will let easily reuse that code in other
+projects and maintains the clarity of our Pipeline.
+In the next post, we'll look at running Sauce OnDemand with
```

Comment:
This last sentence seems kind of crammed in here. I suggest rewording to flow better. (no suggestions off the top of my head)